src/lib.rs

```diff
@@ -26,10 +26,13 @@
 //! In the presence of pathological temporary file cleaner, relying on file paths is unsafe because
 //! a temporary file cleaner could delete the temporary file which an attacker could then replace.
 //!
-//! `tempfile` doesn't rely on file paths so this isn't an issue. However, `NamedTempFile` does
-//! rely on file paths for _some_ operations. See the security documentation on
-//! the `NamedTempFile` type for more information.
-//! 
+//! `tempfile` doesn't rely on file paths, so this should not be an issue. However,
+//! `NamedTempFile` does rely on file paths for _some_ operations. On Unix-like
+//! operating systems and depending on your use case, it may be possible to
+//! mitigate this issue by overriding the crate's default options. For more information,
+//! consult the Security documentation of the [`NamedTempFile`] type,
+//! [`Builder::permissions`] and [`env::override_temp_dir`].
```

Comment:
Note: `Builder::permissions` is relevant for `Builder::tempdir`, but not for `Builder::tempfile` (`NamedTempFile`) because:

1. Changing the permissions of the file won't stop temporary file cleaners (the main issue).
2. The permissions default to 0600.

However, setting the permissions _is_ an important security consideration for temporary _directories_, so that should probably be called out.

Reply:
> The permissions default to 0600.

I presume that this is not the case on Windows (with an equivalent permission)? I sent a revised version.

Reply:
Either way, it would be interesting to document that so that people know how to work around that if necessary in the meantime (e.g. using `/tmp` on Linux, but using a custom directory with specific permissions on Windows) and to allow room for a contribution that fixes the need for such a workaround to be made.

Reply:
Thank you for clarifying. I think I did not really see the forest for the trees. I'll revise my change again for increased clarity.

> a temporary file cleaner could delete the temporary file which an attacker could then replace.

Reply:
I changed the wording re: "predictable file names" without explicitly explaining the problem. I chose to link to the affected / relevant functions instead.